Random number generator

OmnesPotens opened the pull request:
### Summary of Changes
Created a random number generator service which defaults to using an auto-seeded Mersenne Twister (pseudo-random) implementation. This choice was made for its speed and reproducibility; if you use the same seed then it will return the same sequence of numbers.

Added tests for the service.

Added a dependency for random-js, thus an npm install is required.

### Issues Fixed
Fixes [#46](https://github.com/adventurerscodex/adventurerscodex/issues/46)


### Changes Proposed (if any)
N/A


### Screen Shot of Proposed Changes (if UI related)
N/A

Sonictherocketman commented:
Hey @OmnesPotens thanks for the PR! I've got a review of it coming in a bit, (don't worry everything is looking good so far).

It looks like you may have accidentally added some of your editor's project config files. Could you please remove them and update the PR? 

Thanks again.

Sonictherocketman commented on src/charactersheet/services/common/rng_service.js:
```diff
@@ -0,0 +1,109 @@
+import {SharedServiceManager} from './shared_service_manager';
+var Random = require('random-js');
```
Could you change this to the es6 style import statements? 

`import Random from 'random-js';`

OmnesPotens commented:
@Sonictherocketman I've just made the changes you've mentioned. Only the random-js package and its requirements have been added, the previous dependencies have been reverted to what they were before my mistake (sorry about that).

I've removed my editor's project files from the repository and added them to the ignore list, I think I got everything please let me know if I missed anything! Again, sorry about that.

No problem, its always good to have tests to verify functionality.

Please feel free to mention anything else you may have qualms about, I'm effectively an amateur when it comes to javascript and web development so I'm open to any critiques or suggestions you may have.

jkrooskos commented on package-lock.json:
```diff
@@ -19,7 +19,7 @@
       "resolved": "https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz",
-      "integrity": "sha512-nne9/IiQ/hzIhY6pdDnbBtz7DjPTKrY00P/zvPSm5pOFkl6xuGrGnXn/VtTNNfNtAfZ9/1RtehkszU9qcTii0Q==",
+      "integrity": "sha1-+PLIh60Qv2f2NPAFtph/7TF5qsg=",
       "dev": true
```
Hi @OmnesPotens thanks for making the changes! It seems that npm is converting the integrity hash from 512 to 1 for all the packages on your system. Please try to resolve this.

Here are a couple of link regarding the issue:

https://github.com/npm/npm/issues/16938
https://github.com/npm/npm/issues/17749

Maybe this will help:

'Ran into the same issue, fixed it by clearing the cache npm cache clean --force

manually deleting node_modules and then running npm install again.'

OmnesPotens replied:
@jkrooskos um, it looks like the integrity hash for that particular entry was original 1 and -not- 512. I just checked the main develop branch to make sure. I went through the troubleshooting from the links you posted just to make sure and cleaned my npm cache, but it didnt "fix" anything so I'm going to say its as it should be.

OmnesPotens replied:
Updating to the latest npmc experimental build also didnt help to solve the issue after using the aforementioned steps.

jkrooskos replied:
hmm, seems to be a known issue. Just curious, what OS are you working with?

OmnesPotens replied:
I'm running Windows 10 on 2 different machines.

jkrooskos commented:
@OmnesPotens Thanks for the great work!

Sonictherocketman commented:
@OmnesPotens Congrats! You're in our newest [blog post](https://adventurerscodex.com/update/2018/02/28/version-1-6-2.html) 🎉
